src/compiler/factory.ts

```diff
@@ -66,6 +71,16 @@ namespace ts {
         return clone;
     }
 
+    /**
+     * Creates a deep, memberwise clone of a node with no source map location.
+     */
+    /* @internal */
+    export function getSynthesizedDeepClone<T extends Node>(node: T | undefined): T | undefined {
+        return node
+            ? getSynthesizedClone(visitEachChild(node, child => getSynthesizedDeepClone(child), nullTransformationContext))

```

Comment:
No need to write `child => getSynthesizedDeepClone(child)` when just `getSynthesizedDeepClone` would do.